### PR TITLE
feat(autoware_adapi_v1_msgs): add cooperation

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add the messages for [cooperation API](https://github.com/autowarefoundation/autoware-documentation/pull/276).

## Related links

https://github.com/autowarefoundation/autoware-documentation/pull/276

## Tests performed

N/A

## Notes for reviewers

N/A

## Interface changes

See https://github.com/autowarefoundation/autoware-documentation/pull/276.

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_adapi_v1_msgs/planning/msg/PlanningSequence.msg
Comment (mitsudome-r) [inline autoware_adapi_v1_msgs/planning/msg/PlanningSequence.msg:1]: Could you tell me what this message is used for?
I

File: autoware_adapi_v1_msgs/planning/msg/SteeringFactor.msg [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
 uint16 direction
 uint16 status
+string behavior
+string sequence
 string detail
+autoware_adapi_v1_msgs/CooperationStatus[<=1] cooperation
+
+
+
+# deprecated constants for type
```
Comment (mitsudome-r) [inline autoware_adapi_v1_msgs/planning/msg/SteeringFactor.msg:25]: Just out of curiosity, what was the policy for deprecated values in AD API?
We should plan when we actually delete these fields, right?

Comment (isamu-takagi) [reply autoware_adapi_v1_msgs/planning/msg/SteeringFactor.msg:25]: However, in this case, the status of [the planning factor API](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/ad-api/features/planning-factors/) is still under development and not yet released, so I will create a PR to remove it later.

File: autoware_adapi_v1_msgs/planning/msg/PlanningSequence.msg
Comment (isamu-takagi) [reply autoware_adapi_v1_msgs/planning/msg/PlanningSequence.msg:1]: This message defines constants used for `behabior` field in SteeringFactor and VelocityFactor messages.

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**